cmd/kubectl-gke-exec-auth-plugin/provider/provider.go

```diff
@@ -0,0 +1,80 @@
+package provider
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"golang.org/x/oauth2/google"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientauth "k8s.io/client-go/pkg/apis/clientauthentication/v1beta1"
+)
+
+var (
+	// defaultScopes:
+	// - cloud-platform is the base scope to authenticate to GCP.
+	// - userinfo.email is used to authenticate to GKE APIs with gserviceaccount
+	//   email instead of numeric uniqueID.
+	defaultScopes = []string{
+		"https://www.googleapis.com/auth/cloud-platform",
+		"https://www.googleapis.com/auth/userinfo.email"}
+)
+
+// ExecCredential return an object of type ExecCredential which
+// holds a bearer token to authenticate to GKE.
+func ExecCredential() (*clientauth.ExecCredential, error) {
+	token, err := accessToken()
+	if err != nil {
+		return nil, err
+	}
+
+	ec := &clientauth.ExecCredential{
+		TypeMeta: meta.TypeMeta{
+			Kind:       "ExecCredential",
+			APIVersion: "client.authentication.k8s.io/v1beta1",
+		},
+		Status: &clientauth.ExecCredentialStatus{
+			Token: token,
```

Comment:
Should we also include expiration?

Reply:
I skipped it, as this was being invoked for every command. But, added it back, just to maintain existing functionality. Dont want to take any chances in case this gets used in some long running kubectl command.